app/src/main/cpp/skyline/kernel/memory.cpp

```diff
@@ -83,68 +236,50 @@ namespace skyline::kernel {
         if (type != memory::AddressSpaceType::AddressSpace36Bit) {
             base = AllocateMappedRange(baseSize, RegionAlignment, KgslReservedRegionSize, addressSpace.size(), false);
 
-            chunks = {
-                ChunkDescriptor{
-                    .ptr = addressSpace.data(),
-                    .size = static_cast<size_t>(base.data() - addressSpace.data()),
-                    .state = memory::states::Reserved,
-                },
-                ChunkDescriptor{
-                    .ptr = base.data(),
-                    .size = base.size(),
-                    .state = memory::states::Unmapped,
-                },
-                ChunkDescriptor{
-                    .ptr = base.end().base(),
-                    .size = addressSpace.size() - reinterpret_cast<u64>(base.end().base()),
-                    .state = memory::states::Reserved,
-                }};
+            chunks[addressSpace.data()] = ChunkDescriptor{
+                .size = addressSpace.size(),
+                .state = memory::states::Unmapped,
+            };
 
             code = base;
 
         } else {
-            base = AllocateMappedRange(baseSize, 1ULL << 36, KgslReservedRegionSize, addressSpace.size(), false);
-            codeBase36Bit = AllocateMappedRange(0x32000000, RegionAlignment, 0xC000000, 0x78000000ULL + reinterpret_cast<size_t>(addressSpace.data()), true);
-
-            chunks = {
-                ChunkDescriptor{
-                    .ptr = addressSpace.data(),
-                    .size = static_cast<size_t>(codeBase36Bit.data() - addressSpace.data()),
-                    .state = memory::states::Heap,  // We can't use reserved here as rtld uses it to know when to halt memory walking
-                },
-                ChunkDescriptor{
-                    .ptr = codeBase36Bit.data(),
-                    .size = codeBase36Bit.size(),
-                    .state = memory::states::Unmapped,
-                },
-                ChunkDescriptor{
-                    .ptr = codeBase36Bit.end().base(),
-                    .size = static_cast<u64>(base.data() - codeBase36Bit.end().base()),
-                    .state = memory::states::Heap,
-                },
-                ChunkDescriptor{
-                    .ptr = base.data(),
-                    .size = base.size(),
-                    .state = memory::states::Unmapped,
-                },
-                ChunkDescriptor{
-                    .ptr = base.end().base(),
-                    .size = addressSpace.size() - reinterpret_cast<u64>(base.end().base()),
-                    .state = memory::states::Reserved,
-                }};
+            codeBase36Bit = AllocateMappedRange(0x78000000, RegionAlignment, 0x8000000, KgslReservedRegionSize, false);
```

Comment:
Minsize seems too high here, not sure it'll succeed on all devices

Reply:
It's basically required because if you make it any smaller then the end of the stack/tlsIo region won't be allocated